packages/wrapped-keys/src/lib/service-client/utils.ts

```diff
@@ -104,29 +105,40 @@ async function getResponseJson<T>(response: Response): Promise<T | string> {
   }
 }
 
+export function generateRequestId(): string {
+  return Math.random().toString(16).slice(2);
+}
+
 export async function makeRequest<T>({
   url,
   init,
+  requestId,
 }: {
   url: string;
   init: RequestInit;
+  requestId: string;
 }) {
-  console.log('Fetching from URL', url);
-  const response = await fetch(url, { ...init });
+  try {
+    const response = await fetch(url, { ...init });
 
-  if (!response.ok) {
-    const errorMessage = await getResponseErrorMessage(response);
-    throw new Error(`Failed to make request for wrapped key: ${errorMessage}`);
-  }
+    if (!response.ok) {
+      const errorMessage = await getResponseErrorMessage(response);
+      throw new Error(`HTTP(${response.status}): ${errorMessage}`);
+    }
 
-  /**
-   *
-   */
-  const result = await getResponseJson<T>(response);
+    const result = await getResponseJson<T>(response);
 
-  if (typeof result === 'string') {
-    throw new Error(`Unexpected response from wrapped key service: ${result}`);
-  }
+    if (typeof result === 'string') {
+      throw new Error(`HTTP(${response.status}): ${result}`);
+    }
 
-  return result;
+    return result;
+  } catch (e: unknown) {
+    throw new Error(
+      `Request(${requestId}) for wrapped key failed. Error: ${
+        (e as Error).message
+        // @ts-expect-error Unknown, but `cause` is on `TypeError: fetch failed` errors
+      }${e.cause ? ' - ' + e.cause : ''}`
```

Comment:
This should be helpful for identifying the Artillery failures

Reply:
Yup :+1: The HTTP error code plus this ought to give us a clear error reason ❇️